plugins/mobile.js

```diff
@@ -4,22 +4,98 @@ Plugin to capture navigator.connection.type on browsers that support it
 */
 
 (function() {
-	var connection;
-
-	if (typeof navigator === "object") {
-		connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection || navigator.msConnection;
-	}
-
-	if (!connection) {
-		return;
-	}
-	BOOMR.addVar({
-		"mob.ct": connection.type,
-		"mob.bw": connection.bandwidth,
-		"mob.mt": connection.metered
-	});
-
-	if (connection.downlinkMax) {
-		BOOMR.addVar("mob.lm", connection.downlinkMax);
-	}
+	BOOMR = window.BOOMR || {};
+	BOOMR.plugins = BOOMR.plugins || {};
+
+	var impl = {
+		connection: null,
+		param_map: {
+			"type": "ct",
+			"bandwidth": "bw",
+			"metered": "mt",
+			"effectiveType": "etype",
+			"downlinkMax": "lm",
+			"downlink": "dl",
+			"rtt": "rtt",
+			"saveData": "sd"
+		},
+
+		/**
+		 * Get the connection information object if available.
+		 *
+		 * @returns {null|NavigatorNetworkInformation} The network information API data.
+		 */
+		getConnection: function() {
+			if (this.connection) {
+				return this.connection;
+			}
+
+			if (typeof navigator === "object") {
+				this.connection = navigator.connection ||
+					navigator.mozConnection ||
+					navigator.webkitConnection ||
+					navigator.msConnection;
+			}
+
+			return this.connection;
+		},
+
+		/**
+		 * Add listener to update network information when it changes.
+		 *
+		 * @returns {void}
+		 */
+		subscribe: function() {
+			var connection = this.getConnection();
+
+			if (connection.addEventListener) {
+				connection.addEventListener("change", function() {
+					this.setConnectionInformation();
```

Comment:
`this.setConnectionInformation()` throws an error in this context due to the `this` reference.

Reply:
fixed

Reply:
I think `bind` is not compatible with older browsers.  Simplest solution is to just use `impl.setConnectionInformation` instead

Reply:
Yes of course. Fixed!